### DISCUSSION TITLE
Elsa 3.0 - Setting up SQL Server

### DISCUSSION THREAD

**Discussion by vinujosey** [elsa-workflows/elsa-core#5659] [Q&A] [answered]
We are trying configure a  maker checker workflow in out application and trying to migrate the ESLA database tables to our dbcontext. Here is the code in the Program

builder.Services.AddElsa(elsa =>
{
    elsa.UseEntityFrameworkPersistence(ef => ef.UseSqlServer(dbconnection));

});

I have added the following nuget packages

 <PackageReference Include="Elsa" Version="3.1.3" />
 <PackageReference Include="Elsa.EntityFrameworkCore" Version="3.1.3" />
 <PackageReference Include="Elsa.Identity" Version="3.1.3" />
 <PackageReference Include="Elsa.Persistence.EntityFramework.Core" Version="2.14.1" />
 <PackageReference Include="Elsa.Persistence.EntityFramework.SqlServer" Version="2.14.1" />
 <PackageReference Include="Elsa.Scheduling" Version="3.1.3" />
 <PackageReference Include="Elsa.Workflows.Api" Version="3.1.3" />
 <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="8.0.4">
 
 But I still get a compile error for the extension methodUseEntityFrameworkPersistence() . Please let me know which nuget package I am missing.

**Comment by glime-ai** [comment, marked as answer]
To resolve the compile error for the `UseEntityFrameworkPersistence()` extension method, you should ensure that you are using the correct versions of the Elsa packages that are compatible with each other. Since you are using Elsa version `3.1.3`, you should not mix it with packages from the `2.x` series, such as `Elsa.Persistence.EntityFramework.Core` and `Elsa.Persistence.EntityFramework.SqlServer` version `2.14.1`.

You should update your NuGet package references to use the `3.x` series for all Elsa-related packages to ensure compatibility. Here's how you can update your `PackageReference` entries:

```xml
<PackageReference Include="Elsa" Version="3.1.3" />
<PackageReference Include="Elsa.EntityFrameworkCore" Version="3.1.3" />
<PackageReference Include="Elsa.Identity" Version="3.1.3" />
<PackageReference Include="Elsa.Scheduling" Version="3.1.3" />
<PackageReference Include="Elsa.Workflows.Api" Version="3.1.3" />
<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="8.0.4" />
```

Remove the following lines as they refer to version `2.x` which is not compatible with Elsa `3.1.3`:

```xml
<PackageReference Include="Elsa.Persistence.EntityFramework.Core" Version="2.14.1" />
<PackageReference Include="Elsa.Persistence.EntityFramework.SqlServer" Version="2.14.1" />
```

After updating the package references, restore the packages using the following command:

```
dotnet restore
```

Then, you can use the `UseEntityFrameworkPersistence()` method as shown in your code snippet. If you still encounter issues, please ensure that your project targets a compatible .NET version (e.g., .NET 6 or .NET 7) and that you have the latest stable versions of the Elsa packages.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by vinujosey** [comment]
I am still getting the same issue. Is there another nuget package that I need to add or for Elsa 3.1 UseEntityFrameworkPersistence is the correct extension for adding SQL Server ?
